src/software/new_geom/rectangle.h

```diff
@@ -91,5 +93,18 @@ class Rectangle : public ConvexPolygon
      */
     Vector diagonal() const;
 
+    /**
+     * Attempts to move all edges of the rectangle outwards or inwards towards
+     * the centre by an "amount" while maintaining the same location for the center of the
+     * rectangle. The rectangle will not shrink to anything smaller than a point.
+     *
+     * @param amount The amount to expand or shrink the rectangle by on all sides, can be
+     * positive or negative
+     *
+     * @return bool Whether it was possible to expand or shrink the rectangle by
+     * amount requested
```

Comment:
Could we specify the expected behavior if the rectangle can't shrink by the given amount? Will it shrink to a point or remain unchanged (or something else entirely)?

Reply:
It will remain unchanged but yes I'll add that in the comment

Reply:
Fixed in latest